oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/util/ElasticIndexUtils.java

```diff
@@ -46,4 +49,52 @@ public static String idFromPath(@NotNull String path) {
         }
         return path;
     }
+
+    /**
+     * Converts a given byte array (of doubles) to a string of space separated doubles.
+     * @param bytes given byte array
+     * @return string of doubles
+     */
+    public static String toDoubleString(byte[] bytes) {
+        List<Double> a = toDoubles(bytes);
+        StringBuilder builder = new StringBuilder();
+        for (Double d : a) {
+            if (builder.length() > 0) {
+                builder.append(' ');
+            }
+            builder.append(d);
+        }
+        return builder.toString();
+    }
+
+    /**
+     * Converts a given byte array (of doubles) to a list of doubles
+     * @param array given byte array
+     * @return list of doubles
+     */
+    public static List<Double> toDoubles(byte[] array) {
+        int blockSize = Double.SIZE / Byte.SIZE;
+        ByteBuffer wrap = ByteBuffer.wrap(array);
+        int capacity = array.length / blockSize;
+        List<Double> doubles = new ArrayList<>(capacity);
```

Comment:
I think it would make sense to log a warning if the size of the array is unexpected... e.g. size 1, or size 7, or so.

Reply:
Added a warning above. Please verify if that's what you meant.